### PR TITLE
Refactor CLI Project

### PR DESCRIPTION
**Pull request by bartimaeusnek**
+ possible breaking change: ProgramArguments-Singleton was removed (needs to be tested)

### CODE REVIEW

File: Shoko.CLI/Program.cs [hunk 1]
```diff
@@ -1,89 +1,74 @@
...
-                    if (appAssembly != null)
-                    {
-                        config.AddUserSecrets(appAssembly, optional: true);
-                    }
-                }
+    private static void ConfigureServices(IServiceCollection services) => services.AddHostedService<Worker>();
 
-                config.AddEnvironmentVariables();
+#region IConfigurationBuilderExtensions
+    private static IConfigurationBuilder AddCommandLineIfNotNull(this IConfigurationBuilder configurationBuilder)
```
Comment (da3dsoul) [inline Shoko.CLI/Program.cs:52]: Correct me if I'm wrong. This allows the existing command line stuff to work, but doesn't bind to the intermediate model like it used to. Is that right?

File: Shoko.CLI/Worker.cs [hunk 2]
```diff
@@ -1,69 +1,101 @@
...
 
-        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
-        {
-            if (!string.IsNullOrEmpty(Args?.Instance)) ServerSettings.DefaultInstance = Args.Instance;
+    private static void InitInstanceLogger() => ShokoServer.Instance.InitLogger();
+    
+    private static string? GetInstanceFromCommandLineArguments()
+    {
+        const int notFound = -1;
+        var       args     = Environment.GetCommandLineArgs();
```
Comment (da3dsoul) [inline Shoko.CLI/Worker.cs:53]: I personally don't care, but our current formatter settings would nuke that whitespace. I'm less anal about that stuff, as long as it makes sense for readability in the current context.

File: Shoko.CLI/Program.cs (hunk 1 above)
Comment (bartimaeusnek) [reply Shoko.CLI/Program.cs:52]: I'm not quite sure which model youre talking about. If you mean the ``services.AddSingleton(arguments);`` Singleton which did only contain the instance name, then yes, that is gone.

**Comment by da3dsoul** [comment]
I don't have any issues with it. I'll let @Cazzar  be anal about formatting or whatever if he cares to. We haven't made any proper templates for code styles. I'll use whatever cazzar wants if he gives me one. I use Rider's defaults

File: Shoko.CLI/Program.cs (hunk 1 above)
Comment (da3dsoul) [reply Shoko.CLI/Program.cs:52]: Mk that's fine

**Pull request merged**